src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -1199,6 +1219,25 @@ private bool EnsureTlsInsecureIsValid(bool value)
             return value;
         }
 
+        private ClusterConnectionMode? GetEffectiveConnectionMode(ClusterConnectionMode? mode, bool? directConnection, string replicaSet)
```

Comment:
Change `mode` parameter name to `connect` (to match the _connect field it corresponds to).